src/Uno.UWP/Storage/Pickers/FileOpenPicker.iOS.cs

```diff
@@ -26,41 +27,62 @@ public partial class FileOpenPicker
 
 		private async Task<FilePickerSelectedFilesArray> PickFilesAsync(bool multiple, CancellationToken token)
 		{
-			var rootController = UIApplication.SharedApplication?.KeyWindow?.RootViewController;
-			if (rootController == null)
-			{
-				throw new InvalidOperationException("Root controller not initialized yet. FolderPicker invoked too early.");
-			}
-
-			var documentTypes = UTTypeMapper.GetDocumentTypes(FileTypeFilter);
-			using var documentPicker = new UIDocumentPickerViewController(documentTypes, UIDocumentPickerMode.Open);
-			documentPicker.AllowsMultipleSelection = multiple;
-
-			var completionSource = new TaskCompletionSource<NSUrl?[]>();
-
-			documentPicker.OverrideUserInterfaceStyle = CoreApplication.RequestedTheme == SystemTheme.Light ?
-				UIUserInterfaceStyle.Light : UIUserInterfaceStyle.Dark;
+			UIViewController viewController = new UIViewController();
 
-			documentPicker.ShouldShowFileExtensions = true;
-			documentPicker.Delegate = new FileOpenPickerDelegate(completionSource);
-
-			if (documentPicker.PresentationController != null)
+			try
 			{
-				documentPicker.PresentationController.Delegate = new FileOpenPickerPresentationControllerDelegate(completionSource);
-			}
-
-			await rootController.PresentViewControllerAsync(documentPicker, true);
+				var rootController = UIApplication.SharedApplication?.KeyWindow?.RootViewController;
+				if (rootController == null)
+				{
+					throw new InvalidOperationException("Root controller not initialized yet. FolderPicker invoked too early.");
+				}
+
+				var documentTypes = UTTypeMapper.GetDocumentTypes(FileTypeFilter);
+
+				switch (SuggestedStartLocation)
+				{
+					case PickerLocationId.PicturesLibrary:
+						viewController = new UIImagePickerController();
+						break;
+					default:
+						viewController = new UIDocumentPickerViewController(documentTypes, UIDocumentPickerMode.Open);
+						break;
+				}
+				var completionSource = new TaskCompletionSource<NSUrl?[]>();
+
+				if (viewController is UIDocumentPickerViewController documentPicker)
+				{
+					documentPicker.AllowsMultipleSelection = multiple;
+					documentPicker.ShouldShowFileExtensions = true;
+					documentPicker.Delegate = new FileOpenPickerDelegate(completionSource);
+				}
```

Comment:
Instead of this being in an if condition, would it make sense to move inside the switch statement under the `default` arm?

It would be as this:

```csharp
viewController = new UIDocumentPickerViewController(documentTypes, UIDocumentPickerMode.Open)
{
	documentPicker.AllowsMultipleSelection = multiple,
	documentPicker.ShouldShowFileExtensions = true,
	documentPicker.Delegate = new FileOpenPickerDelegate(completionSource),
};
```

and completionSource will have to be moved before the switch statement.